app/models/concerns/host_common.rb

```diff
@@ -117,7 +117,17 @@ def crypt_root_pass
                        end
 
     if unencrypted_pass.present?
-      is_actually_encrypted = if PasswordCrypt.crypt_gnu_compatible?
+      is_actually_encrypted = if operatingsystem.try(:password_hash) == "Base64"
```

Comment:
Can you refactor this into a few methods? It's very complicated to know what's going on, especially with this level of nesting.